tests/alpaka/alpaka_main.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Alpaka include(s).
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+
+// GoogleTest include(s).
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <iostream>
+
+template <typename Acc>
+ALPAKA_FN_ACC float process(Acc const& acc, uint32_t idx) {
+    return alpaka::math::sin(acc, idx);
+}
+
+struct VectorOpKernel {
+    template <typename Acc>
+    ALPAKA_FN_ACC void operator()(Acc const& acc, float* result,
+                                  uint32_t n) const {
+        using namespace alpaka;
+        auto const globalThreadIdx =
+            getIdx<alpaka::Grid, alpaka::Threads>(acc)[0u];
+
+        // It is possible to get index type from the accelerator,
+        // but for simplicity we just repeat the type here
+        if (globalThreadIdx < n) {
+            result[globalThreadIdx] = process(acc, globalThreadIdx);
+        }
+    }
+};
```

Comment:
:thinking: For this simple test I would rather just go with a simple function.

```c++
template <typename Acc>
void VectorOpKernel(Acc const& acc, float* result, uint32_t n) {...}
```

Reply:
Scratch this one. Alpaka is not as flexible with "function objects" as SYCL... :frowning:

```
[100%] Building CXX object tests/alpaka/CMakeFiles/traccc_test_alpaka.dir/alpaka_main.cpp.o
/home/krasznaa/ATLAS/projects/traccc/traccc/tests/alpaka/alpaka_main.cpp:68:5: error: no matching function for call to 'exec'
    alpaka::exec<Acc>(queue, workDiv, VectorOpKernel,
    ^~~~~~~~~~~~~~~~~
/home/krasznaa/ATLAS/projects/traccc/build/_deps/alpaka-src/include/alpaka/kernel/Traits.hpp:269:25: note: candidate template ignored: couldn't infer template argument 'TKernelFnObj'
    ALPAKA_FN_HOST auto exec(TQueue& queue, TWorkDiv const& workDiv, TKernelFnObj const& kernelFnObj, TArgs&&... args)
                        ^
1 error generated.
make[2]: *** [tests/alpaka/CMakeFiles/traccc_test_alpaka.dir/build.make:76: tests/alpaka/CMakeFiles/traccc_test_alpaka.dir/alpaka_main.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:4987: tests/alpaka/CMakeFiles/traccc_test_alpaka.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```